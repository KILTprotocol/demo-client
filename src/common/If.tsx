import * as React from 'react'

type Props = {
  condition: boolean
  then: any
  else?: any
}

const If: React.FunctionComponent<Props> = props => {
  if (props.condition) {
    return props.then
  }

  if (!!props.else) {
    return props.else
  }

  return null
}

export default If
