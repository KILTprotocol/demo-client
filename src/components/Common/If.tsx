type Props = {
  condition: boolean
  then: any,
  else?: any
}

const If: React.SFC<Props> = props => {
  if (props.condition) {
    return props.then
  } else if (!!props.else) {
    return props.else
  }
  return null
}

export default If