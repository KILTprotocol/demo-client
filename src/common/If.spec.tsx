import * as React from 'react'
import * as Renderer from 'react-test-renderer'
import If from './If'

describe('If Component', () => {
  it('renders true condition', () => {
    const component = Renderer.create(<If condition={true} then={<div>Then</div>} else={<div>Else</div>} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders false condition', () => {
    const component = Renderer.create(<If condition={false} then={<div>Then</div>} else={<div>Else</div>} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders true condition without else', () => {
    const component = Renderer.create(<If condition={true} then={<div>Then</div>} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders false condition without else', () => {
    const component = Renderer.create(<If condition={false} then={<div>Then</div>} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
