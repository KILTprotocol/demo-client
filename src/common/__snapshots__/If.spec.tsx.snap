// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If Component renders false condition 1`] = `
<div>
  Else
</div>
`;

exports[`If Component renders false condition without else 1`] = `null`;

exports[`If Component renders true condition 1`] = `
<div>
  Then
</div>
`;

exports[`If Component renders true condition without else 1`] = `
<div>
  Then
</div>
`;
